apps/service_providers/llm_service/main.py

```diff
@@ -151,31 +161,42 @@ def attach_built_in_tools(self, built_in_tools: list[str], config: dict[str, Bas
     def get_output_parser(self):
         return self._default_parser
 
-    def _default_parser(self, output):
-        output = output.get("output", "")
-        if isinstance(output, list):
-            return "\n".join([o["text"] for o in output])
-
-        return output or ""
-
-    def get_cited_files_parser(self):
-        return self._default_cited_files_parser
-
-    def _default_cited_files_parser(self, token: str | dict, team_id: int) -> list[File]:
-        remote_file_ids = []
-        file_ids = []
-        if isinstance(token, dict):
-            outputs = token.get("output", "")
-            if isinstance(outputs, list):
-                for output in outputs:
-                    annotation_entries = output.get("annotations", [])
-                    external_ids = [entry["file_id"] for entry in annotation_entries if "file_id" in entry]
-                    external_ids = detangle_file_ids(external_ids)
-                    remote_file_ids.extend(external_ids)
-            else:
-                file_ids.extend(extract_file_ids_from_ocs_citations(outputs))
+    def _default_parser(
+        self, llm_output, session: ExperimentSession, include_citations: bool = True
+    ) -> LlmChatResponse:
+        llm_outputs = llm_output.get("output", "")
+        final_text = ""
+        cited_file_ids_remote = []
+        cited_file_ids = []
+        cited_files = []
+        generated_files: list[File] = []
+        if isinstance(llm_outputs, list):
```

Comment:
This seems to indicate that the output was from a different API or provider, is that correct?